tests/stdlib/tarithmetics.nim

```diff
@@ -0,0 +1,44 @@
+discard """
+  targets: "c cpp js"
+"""
+
+# TODO: in future work move existing arithmetic tests here, where they belong: Nim/tests/arithm
+# FYI https://github.com/nim-lang/Nim/pull/17767
+
+template main =
+  # put all arithmetic tests
+
+  block tshr:
+
+    # Signed types
+    block:
+      let
+        t0: int = -3 shr 2
```

Comment:
https://github.com/timotheecour/Nim/issues/534#issuecomment-780791150

Reply:
this advise is maybe following ?

brefore
``` nim
let t0: int = -3 shr 2
doAssert t0 == -1
```

after
``` nim
doAssert -3 shr 2 == -1
```

Reply:
that has been resolved